cara/apps/calculator/__init__.py

```diff
@@ -222,6 +254,7 @@ def make_app(
         (r'/static/(.*)', StaticFileHandler, {'path': static_dir}),
         (calculator_prefix + r'/?', CalculatorForm),
         (calculator_prefix + r'/report', ConcentrationModel),
+        (calculator_prefix + r'/report-json', ConcentrationModelJsonResponse),
```

Comment:
Normally I would suggest the return type doesn't belong in the URL, rather it is a request parameter (``content-type``). Is there a good reason for making it this way?

Reply:
One reason, may be: we should be using the GET endpoint, with input JSON as an input argument. This would mean we also don't need to be able to toggle the XSRF cookies as well.

Reply:
I changed the endpoint now from POST to GET. Indeed, this solves the issues with the XSRF cookie. As I understand, it's not standard practice to send a request body with a HTTP GET method. But it's not "forbidden" either. In our case, defining an additional GET endpoint might be the better solution.

Reply:
> it's not standard practice to send a request body with a HTTP GET method

Agreed (https://stackoverflow.com/a/983458/741316). I actually meant passing it as a query string to a GET request. This however has its own downsides (query string limit in practice is ~1024 chars), which may mean that we can't do that. In order of preference:

```
GET + querystring > POST + body + xsrf disabled > GET + body
```

If we need to go back to POST then we can disable XSRF on a per-endpoint basis by implementing the ``ConcentrationModelJsonResponse.check_xsrf_cookie`` method (https://github.com/tornadoweb/tornado/issues/254#issuecomment-147694082).

Reply:
I think I would prefer `POST + body + xsrf disabled` or `GET + body` over `GET + querystring`. For the same reasons you mention in your comment (querry string limit). In that case, I think the `POST + body + xsrf disabled` would be the preferred solution. Should I go back to the POST trying to disabling XSRF on a per-endpoint basis as you suggested?

Reply:
Yes please, this would be best (POST + body + per-endpoint XSRF disable). Apologies for the flip-flop.